src/mbgl/mtl/texture2d.cpp

```diff
@@ -75,171 +71,117 @@ size_t Texture2D::numChannels() const noexcept {
     }
 }
 
-/*
-void Texture2D::createObject() noexcept {
+MTL::PixelFormat Texture2D::getMetalPixelFormat() const noexcept {
+    switch (channelType) {
+        case gfx::TextureChannelDataType::UnsignedByte:
+            switch (pixelFormat) {
+                case gfx::TexturePixelType::RGBA:
+                    return MTL::PixelFormat::PixelFormatRGBA8Unorm;
+                case gfx::TexturePixelType::Alpha:
+                    return MTL::PixelFormat::PixelFormatA8Unorm;
+                default:
+                    assert(false);
+                    throw std::runtime_error("Pixel format not supported on Metal!");
+            }
+        case gfx::TextureChannelDataType::HalfFloat:
+            switch (pixelFormat) {
+                case gfx::TexturePixelType::RGBA:
+                    return MTL::PixelFormat::PixelFormatRGBA16Float;
+                case gfx::TexturePixelType::Alpha:
+                    return MTL::PixelFormat::PixelFormatR16Float;
+                default:
+                    assert(false);
+                    throw std::runtime_error("Pixel format not supported on Metal!");
+            }
+    }
+}
```

Comment:
Same here, I used linear color space.

Reply:
I think Apple recommends using sRGB where possible, but this could lead to testing variance even if the result is subjectively better.  Perhaps in a later version this could be an API option.

Reply:
Yes, we can adjust later on, when reaching the render test phase.